### PR TITLE
Usr/abhishek sinha10/molecule enhancement 870

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
# Description
Enhancing Molecule code for import_server_config_profile, idrac_server_powerstate and redfish_firmware.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
import_server_config_profile
idrac_server_powerstate
redfish_firmware

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: roles/idrac_server_powerstate/molecule/default/converge.yml [hunk 1]
```diff
@@ -141,16 +141,10 @@
...
-      ansible.builtin.set_fact:
-        resource_id_list: "{{ uri_out.json | json_query(query) }}"
-      vars:
-        query: Members[*]."@odata.id"
-
     - name: Splitting the resource_id and picking last
       ansible.builtin.set_fact:
-        resource_id_list_split: "{{ [item | split('/') | last] + resource_id_list_split | default([]) }}"
-      loop: "{{ resource_id_list }}"
+        resource_id_list_split: "{{ [item['@odata.id'] | split('/') | last] + resource_id_list_split | default([]) }}"
```
Comment (sachin-apa) [inline roles/idrac_server_powerstate/molecule/default/converge.yml]: To keep it in sync can we rename this fact and register variables to role_name_

File: roles/idrac_server_powerstate/molecule/forcerestart/converge.yml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 - name: Converge
   hosts: all
   gather_facts: false
+  vars:
+    wait_seconds: 180
```
Comment (sachin-apa) [inline roles/idrac_server_powerstate/molecule/forcerestart/converge.yml]: To keep it in sync can we rename this fact and register variables to role_name_

File: roles/idrac_server_powerstate/molecule/nmi/converge.yml [hunk 3]
```diff
@@ -2,6 +2,8 @@
 - name: Converge
   hosts: all
   gather_facts: false
+  vars:
+    wait_seconds: 180
```
Comment (sachin-apa) [inline roles/idrac_server_powerstate/molecule/nmi/converge.yml]: To keep it in sync can we rename this fact and register variables to role_name_

File: roles/idrac_server_powerstate/molecule/pushpowerbutton/converge.yml [hunk 4]
```diff
@@ -2,6 +2,8 @@
 - name: Converge
   hosts: all
   gather_facts: false
+  vars:
+    wait_seconds: 180
```
Comment (sachin-apa) [inline roles/idrac_server_powerstate/molecule/pushpowerbutton/converge.yml]: To keep it in sync can we rename this fact and register variables to role_name_

**Review by sachin-apa — COMMENTED** [review]
lets please check all the files for facts and register variables and rename to role_name_  it will be a future proof ..

File: roles/idrac_server_powerstate/molecule/default/converge.yml (hunk 1 above)
Comment (ABHISHEK-SINHA10) [reply roles/idrac_server_powerstate/molecule/default/converge.yml]: Sure

File: roles/idrac_server_powerstate/molecule/forcerestart/converge.yml (hunk 2 above)
Comment (ABHISHEK-SINHA10) [reply roles/idrac_server_powerstate/molecule/forcerestart/converge.yml]: Sure

File: roles/idrac_server_powerstate/molecule/pushpowerbutton/converge.yml (hunk 4 above)
Comment (ABHISHEK-SINHA10) [reply roles/idrac_server_powerstate/molecule/pushpowerbutton/converge.yml]: Sure

File: roles/idrac_server_powerstate/molecule/nmi/converge.yml (hunk 3 above)
Comment (ABHISHEK-SINHA10) [reply roles/idrac_server_powerstate/molecule/nmi/converge.yml]: Sure

File: roles/molecule.yml [hunk 5]
```diff
@@ -13,11 +13,10 @@ platforms:
 provisioner:
   name: ansible
   env:
-    RUNON: ""
     idrac_certificate_delegate_to: localhost
-    idrac_server_powerstate_wait_seconds: "180"
```
Comment (Kritika-Bhateja-03) [inline roles/molecule.yml:18]: Why are we not defining the wait time here instead of defining it in each of converge files?

Comment (ABHISHEK-SINHA10) [reply roles/molecule.yml:18]: The reason I want to keep only important info in molecule.yml and remove which is minor.

Comment (ABHISHEK-SINHA10) [reply roles/molecule.yml:18]: I'm resolving this, if not satisfied, ping me offline, we can discuss.

**Review by jagadeeshnv — APPROVED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Pull request closed without merging**